alerting/src/main/kotlin/org/opensearch/alerting/settings/SupportedClusterMetricsSettings.kt

```diff
@@ -0,0 +1,138 @@
+
+            @Suppress("UNCHECKED_CAST")
+            if (supportedJsonPayloads != null)
+                supportedApiList = XContentHelper.convertToMap(JsonXContent.jsonXContent, supportedJsonPayloads.readText(), false) as HashMap<String, Map<String, ArrayList<String>>>
+        }
+
+        /**
+         * Returns the map of all supported json payload associated with the provided path from supportedApiList.
+         * @param path The path for the requested API.
+         * @return The map of the supported json payload for the requested API.
+         * @throws IllegalArgumentException When supportedApiList does not contain a value for the provided key.
+         */
+        fun getSupportedJsonPayload(path: String): Map<String, ArrayList<String>> {
+            return supportedApiList[path] ?: throw IllegalArgumentException("API path not in supportedApiList.")
+        }
+
+        /**
+         * Will return an [ActionRequest] for the API associated with that path.
+         * Will otherwise throw an exception.
+         * @param clusterMetricsInput The [ClusterMetricsInput] to resolve.
+         * @throws IllegalArgumentException when the requested API is not supported.
+         * @return The [ActionRequest] for the API associated with the provided [ClusterMetricsInput].
+         */
+        fun resolveToActionRequest(clusterMetricsInput: ClusterMetricsInput): ActionRequest {
+            val pathParams = clusterMetricsInput.parsePathParams()
+            return when (clusterMetricsInput.clusterMetricType) {
+                ClusterMetricType.CAT_PENDING_TASKS -> PendingClusterTasksRequest()
+                ClusterMetricType.CAT_RECOVERY -> {
+                    if (pathParams.isEmpty()) return RecoveryRequest()
+                    val pathParamsArray = pathParams.split(",").toTypedArray()
+                    return RecoveryRequest(*pathParamsArray)
+                }
+                ClusterMetricType.CAT_SNAPSHOTS -> {
+                    return GetSnapshotsRequest(pathParams, arrayOf(GetSnapshotsRequest.ALL_SNAPSHOTS))
+                }
+                ClusterMetricType.CAT_TASKS -> ListTasksRequest()
+                ClusterMetricType.CLUSTER_HEALTH -> {
+                    if (pathParams.isEmpty()) return ClusterHealthRequest()
+                    val pathParamsArray = pathParams.split(",").toTypedArray()
+                    return ClusterHealthRequest(*pathParamsArray)
+                }
+                ClusterMetricType.CLUSTER_SETTINGS -> ClusterStateRequest().routingTable(false).nodes(false)
+                ClusterMetricType.CLUSTER_STATS -> {
+                    if (pathParams.isEmpty()) return ClusterStatsRequest()
+                    val pathParamsArray = pathParams.split(",").toTypedArray()
+                    return ClusterStatsRequest(*pathParamsArray)
+                }
+                ClusterMetricType.NODES_STATS -> NodesStatsRequest().addMetrics(
```

Comment:
this is good for now, but ideally, we want to give control to choose what metrics or in general params they would want to pass. Let's add a todo task here.

Reply:
Created this issue to track this feedback https://github.com/opensearch-project/alerting/issues/341